sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -809,6 +794,17 @@ public ResourceId apply(String input) {
     }
 
     /**
+     * Returns a transform for writing to text files like this one but that compresses output using
+     * the given {@link Compression}. The default is value is {@link Compression#UNCOMPRESSED}.
+     *
+     * <p>A {@code null} value will reset the value to the default value mentioned above.
+     */
```

Comment:
<!--new_thread; commit:12876d43a961dd8ff9dba5f4cffcfbe643927eee; resolved:0-->
Passing in null into FileBasedSink.CompressionType.fromCanonical will throw an exception.

Is there a reason why you want to treat `null` as use default?

Reply:
Done, disallowing null in the new method.